src/Decorators/EngineDecorator.php

```diff
@@ -6,13 +6,19 @@
 use ElasticAdapter\Search\SearchRequest;
 use ElasticAdapter\Search\SearchResponse;
 use ElasticScoutDriver\Engine;
+use ElasticScoutDriver\Factories\DocumentFactoryInterface;
+use ElasticScoutDriverPlus\ShardRouting;
 use ElasticScoutDriverPlus\Support\ModelScope;
 use Illuminate\Support\Traits\ForwardsCalls;
 
 final class EngineDecorator
```

Comment:
Maybe we should now better extend the Engine::class?

Reply:
Alright will need to open another pr on scout-driver to make Engine non final and change the fields to protected instead of private.

Reply:
https://github.com/babenkoivan/elastic-scout-driver/pull/21